Argo/Types/Decoded.swift

```diff
@@ -12,14 +12,23 @@ public enum Decoded<T> {
     default: return .None
     }
   }
+}
 
-  public static func optional<A>(p: Decoded<A>) -> Decoded<A?> {
-    switch p {
+public extension Decoded {
+  static func optional<T>(x: Decoded<T>) -> Decoded<T?> {
```

Comment:
I don't have a better name to recommend, I'm ok with this and was just seeing if other people had ideas for it.